### PR TITLE
Fix #10404 - Multiple arguments support for db, db-, dbw, removed dbwC (undocumented duplicate of dbC)

### LINKED ISSUE #10404: All db commands must accept multiple arguments
We must be able to set, enable/disable/toggle multiple breakpoints in one line

### PR DESCRIPTION
**Pull request by NotSoOld**
Heavily tested those, can be safely merged.
dbwC did the same thing as dbC does, also tested that fact.

### CODE REVIEW

File: libr/core/cmd_debug.c [hunk 1]
```diff
@@ -3370,94 +3381,71 @@ static void r_core_cmd_bp(RCore *core, const char *input) {
...
-							if (addr > f->offset) {
-								bpi->name = r_str_newf ("%s+0x%" PFMT64x, f->name, addr - f->offset);
+		char *str = strdup (p);
+		int sl = r_str_word_set0 (str);
+		// For dbw every second argument is 'rw', so we need to skip it.
+		for (int i = 0; i < sl; i += 1 + (watch ? 1 : 0)) {
+			if (*DB_ARG(i) == '-') {
+				r_bp_del (core->dbg->bp, r_num_math (core->num, DB_ARG(i) + 1));
+			} else {
+        if (watch) {
```
Comment (ret2libc) [inline libr/core/cmd_debug.c]: fix indentation please

File: libr/core/cmd_debug.c [hunk 2]
```diff
@@ -3370,94 +3381,72 @@ static void r_core_cmd_bp(RCore *core, const char *input) {
...
+		for (int i = 0; i < sl; i += 1 + (watch ? 1 : 0)) {
+			if (*DB_ARG(i) == '-') {
+				r_bp_del (core->dbg->bp, r_num_math (core->num, DB_ARG(i) + 1));
+			} else {
+				if (watch) {
+					if (sl % 2 == 0) {
+						if (!strcmp (DB_ARG(i + 1), "r")){
+							rw = R_BP_PROT_READ;
+						}
+						else if (!strcmp (DB_ARG(i + 1), "w")) {
```
Comment (radare) [inline libr/core/cmd_debug.c]: else goes in the same line as }

File: libr/core/cmd_debug.c [hunk 3]
```diff
@@ -3370,94 +3381,72 @@ static void r_core_cmd_bp(RCore *core, const char *input) {
...
+		char *str = strdup (p);
+		int sl = r_str_word_set0 (str);
+		// For dbw every second argument is 'rw', so we need to skip it.
+		for (int i = 0; i < sl; i += 1 + (watch ? 1 : 0)) {
+			if (*DB_ARG(i) == '-') {
+				r_bp_del (core->dbg->bp, r_num_math (core->num, DB_ARG(i) + 1));
+			} else {
+				if (watch) {
+					if (sl % 2 == 0) {
+						if (!strcmp (DB_ARG(i + 1), "r")){
```
Comment (radare) [inline libr/core/cmd_debug.c]: please honor indentation rules. add a space before every (

File: libr/core/cmd_debug.c [hunk 4]
```diff
@@ -3370,94 +3381,72 @@ static void r_core_cmd_bp(RCore *core, const char *input) {
...
-			if (validAddress (core, addr)) {
-				bpi = r_debug_bp_add (core->dbg, addr, hwbp, watch, rw, NULL, 0);
-				if (bpi) {
-					free (bpi->name);
-					if (!strcmp (input + 2, "$$")) {
-						RFlagItem *f = r_flag_get_i2 (core->flags, addr);
-						if (f) {
-							if (addr > f->offset) {
-								bpi->name = r_str_newf ("%s+0x%" PFMT64x, f->name, addr - f->offset);
+				addr = r_num_math (core->num, DB_ARG(i));
```
Comment (radare) [inline libr/core/cmd_debug.c]: here too

File: libr/core/cmd_debug.c [hunk 5]
```diff
@@ -3370,94 +3381,72 @@ static void r_core_cmd_bp(RCore *core, const char *input) {
...
+								"See e? dbg.bpinmaps\n", addr);
 				}
-			} else {
-				eprintf ("Cannot place a breakpoint on 0x%08"PFMT64x" unmapped memory. See e? dbg.bpinmaps\n", addr);
 			}
-			free (str);
 		}
+		free (str);
 		break;
+		#undef DB_ARG
```
Comment (radare) [inline libr/core/cmd_debug.c]: i dont think that works with macros

File: libr/core/cmd_debug.c [hunk 6]
```diff
@@ -3243,8 +3243,19 @@ static void r_core_cmd_bp(RCore *core, const char *input) {
 	case '*': r_bp_list (core->dbg->bp, 1); break;
 	case '\0': r_bp_list (core->dbg->bp, 0); break;
 	case '-': // "db-"
-		if (input[2] == '*') r_bp_del_all (core->dbg->bp);
-		else r_bp_del (core->dbg->bp, r_num_math (core->num, input + 2));
+		if (input[2] == '*') {
+			r_bp_del_all (core->dbg->bp);
+		} else {
+			#define DB_ARG(x) r_str_word_get0(str, x)
+			for (p = input + 2; *p == ' '; p++);
```
Comment (radare) [inline libr/core/cmd_debug.c]: better use char *str = strdup (r_str_trim_ro (input +2));

File: libr/core/cmd_debug.c [hunk 7]
```diff
@@ -3243,8 +3243,19 @@ static void r_core_cmd_bp(RCore *core, const char *input) {
...
+			char *str = strdup (p);
+			int sl = r_str_word_set0 (str);
+			for (int i = 0; i < sl; i++) {
+				r_bp_del (core->dbg->bp, r_num_math (core->num, DB_ARG(i)));
```
Comment (radare) [inline libr/core/cmd_debug.c]: maybe better split this line in 2, const ut64 addr = r_num_math... and then rbpdel(addr)

Comment (NotSoOld) [reply libr/core/cmd_debug.c]: Thanks for review, I will fix these things, but, honestly speaking, most of these lines already were in the code in such condition, so it's not my fault.

**Comment by XVilka** [comment]
Ignore the PDB downloader failed test - Travis is essentially green.

**Pull request merged**